linkerd/app/inbound/src/direct.rs

```diff
@@ -125,18 +145,11 @@ impl<N> Inbound<N> {
                                     // it.
                                     let addr = (client.local_addr.ip(), port).into();
                                     let allow = policies.check_policy(OrigDstAddr(addr))?;
-                                    let tls = tls::ConditionalServerTls::Some(
-                                        tls::ServerTls::Established {
-                                            client_id: Some(client.client_id.clone()),
-                                            negotiated_protocol: client.alpn,
-                                        },
-                                    );
-                                    let permit =
-                                        allow.check_authorized(client.client_addr, &tls)?;
```

Comment:
If I understand correctly, we are no longer checking the policy on opaque connections?

It's probably better to introduce a new target type--`LocalHttp` or something (and rename `Local` to `LocalTcp`?). This switch could return an `Either<Either<LocalTcp, LocalHttp>, GatewayTransportHeader>` and then the inner switch predicate can simply return the target... I can probably put a suggestion up to this effect.

Reply:
https://github.com/linkerd/linkerd2-proxy/pull/1417